src/components/MessageButton.ts

```diff
@@ -0,0 +1,281 @@
+import type { APIPartialEmoji, Snowflake } from 'discord-api-types/v8';
+import ow from 'ow';
+import { APIBaseComponent, ComponentType } from './BaseComponent';
+
+/**
+ * Builds a button-type message component
+ */
+export class MessageButton {
+	/**
+	 * The text to be displayed on this button
+	 * @type {?string}
+	 */
+	public customID?: string;
+
+	/**
+	 * Whether this button is currently disabled
+	 * @type {boolean}
+	 */
+	public disabled?: boolean;
+
+	/**
+	 * Emoji for this button
+	 * @type {?RawEmoji}
+	 */
+	public emoji?: APIPartialEmoji;
+
+	/**
+	 * The text to be displayed on this button
+	 * @type {?string}
+	 */
+	public label?: string;
+
+	/**
+	 * The style of this button
+	 * @type {?MessageButtonStyle}
+	 */
+	public style?: MessageButtonStyle;
+
+	/**
+	 * The URL this button links to, if it is a Link style button
+	 * @type {?string}
+	 */
+	public url?: string;
+
+	/**
+	 * @param {MessageButtonData|APIButtonComponent} [data] MessageButton to clone or raw data
+	 */
+	public constructor(data?: MessageButtonData | APIButtonComponent) {
+		if (!data) return;
+
+		this.style = MessageButton.resolveStyle(data.style);
+
+		if ('custom_id' in data && typeof data.custom_id !== 'undefined') {
+			this.setCustomID(data.custom_id);
+		} else if ('customID' in data && typeof data.customID !== 'undefined') {
+			this.setCustomID(data.customID);
+		}
+
+		if (typeof data.label !== 'undefined') {
+			this.setLabel(data.label);
+		}
+
+		if (typeof data.emoji !== 'undefined') {
+			this.setEmoji(data.emoji);
+		}
+
+		if (typeof data.url !== 'undefined') {
+			this.setURL(data.url);
+		}
+
+		this.setDisabled(data.disabled ?? false);
+	}
+
+	/**
+	 * Sets the custom ID of this button
+	 * @param {string} customID A unique string to be sent in the interaction when clicked
```

Comment:
```suggestion
	 * @param customId A unique string to be sent in the interaction when clicked
```

In TS we don't need to document param types.